<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ModbusTool</class>
 <widget class="QMainWindow" name="ModbusTool">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>542</width>
    <height>487</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ModbusTool</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>524</width>
        <height>442</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string/>
      </property>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="2" column="1">
        <widget class="QTextEdit" name="txtMessage">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="1" rowspan="2">
        <widget class="QGroupBox" name="groupBox_8">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="title">
          <string>常用指令</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <widget class="QPushButton" name="btn03Send">
            <property name="text">
             <string>03H</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" rowspan="3">
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="0" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>从地址</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>寄存器地址</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>寄存器数量</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLineEdit" name="txt03SlaveAddr"/>
            </item>
            <item row="1" column="1">
             <widget class="QLineEdit" name="txt03RegAddr"/>
            </item>
            <item row="1" column="2">
             <widget class="QLineEdit" name="txt03RegNum"/>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>从地址</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>寄存器地址</string>
              </property>
             </widget>
            </item>
            <item row="2" column="2">
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>寄存器数量</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLineEdit" name="txt04SlaveAddr"/>
            </item>
            <item row="3" column="1">
             <widget class="QLineEdit" name="txt04RegAddr"/>
            </item>
            <item row="3" column="2">
             <widget class="QLineEdit" name="txt04RegNum"/>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>从地址</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>寄存器地址</string>
              </property>
             </widget>
            </item>
            <item row="4" column="2">
             <widget class="QLabel" name="label_12">
              <property name="text">
               <string>寄存器值</string>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="QLineEdit" name="txt06SlaveAddr"/>
            </item>
            <item row="5" column="1">
             <widget class="QLineEdit" name="txt06RegAddr"/>
            </item>
            <item row="5" column="2">
             <widget class="QLineEdit" name="txt06Value"/>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="btn04Send">
            <property name="text">
             <string>04H</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QPushButton" name="btn06Send">
            <property name="text">
             <string>06H</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QPushButton" name="btn10Send">
            <property name="text">
             <string>10H</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="0" column="0">
             <widget class="QLabel" name="label_13">
              <property name="minimumSize">
               <size>
                <width>35</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>35</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>从地址</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLabel" name="label_15">
              <property name="text">
               <string>寄存器地址</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLabel" name="label_16">
              <property name="text">
               <string>寄存器数量</string>
              </property>
             </widget>
            </item>
            <item row="0" column="3">
             <widget class="QLabel" name="label_14">
              <property name="text">
               <string>数据值</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLineEdit" name="txt10SlaveAddr">
              <property name="minimumSize">
               <size>
                <width>35</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>35</width>
                <height>16777215</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLineEdit" name="txt10RegAddr"/>
            </item>
            <item row="1" column="2">
             <widget class="QLineEdit" name="txt10RegNum"/>
            </item>
            <item row="1" column="3">
             <widget class="QLineEdit" name="txt10Value"/>
            </item>
           </layout>
          </item>
          <item row="4" column="0">
           <widget class="QPushButton" name="btnOtherSend">
            <property name="text">
             <string>其他指令</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="txtOtherValue"/>
          </item>
         </layout>
         <zorder></zorder>
         <zorder>btn10Send</zorder>
         <zorder></zorder>
         <zorder></zorder>
         <zorder></zorder>
         <zorder>splitter_2</zorder>
         <zorder>splitter_3</zorder>
         <zorder>txt06Value</zorder>
         <zorder>txt06RegAddr</zorder>
         <zorder>txt06SlaveAddr</zorder>
         <zorder>btn06Send</zorder>
         <zorder>label_9</zorder>
         <zorder>label_11</zorder>
         <zorder>label_12</zorder>
         <zorder>label_9</zorder>
         <zorder>label_11</zorder>
         <zorder>txt04RegNum</zorder>
         <zorder>btn04Send</zorder>
         <zorder>txt04RegAddr</zorder>
         <zorder>txt04SlaveAddr</zorder>
         <zorder>label_7</zorder>
         <zorder>label_4</zorder>
         <zorder>label_8</zorder>
         <zorder>txt03RegNum</zorder>
         <zorder>txt03SlaveAddr</zorder>
         <zorder>btn03Send</zorder>
         <zorder>txt03RegAddr</zorder>
         <zorder>label_3</zorder>
         <zorder>label</zorder>
         <zorder>label_2</zorder>
         <zorder>splitter_4</zorder>
        </widget>
       </item>
       <item row="1" column="0" rowspan="2">
        <widget class="QGroupBox" name="groupBox_2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="title">
          <string>Config</string>
         </property>
         <widget class="QCheckBox" name="checkBox">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>57</y>
            <width>91</width>
            <height>18</height>
           </rect>
          </property>
          <property name="text">
           <string>带CRC校验码</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QRadioButton" name="robMaster">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>81</y>
            <width>43</width>
            <height>18</height>
           </rect>
          </property>
          <property name="text">
           <string>RTU</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">bgpMSS</string>
          </attribute>
         </widget>
         <widget class="QRadioButton" name="robSlaver">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>105</y>
            <width>55</width>
            <height>18</height>
           </rect>
          </property>
          <property name="text">
           <string>ASCII</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">bgpMSS</string>
          </attribute>
         </widget>
         <widget class="QToolButton" name="btnSubtract">
          <property name="geometry">
           <rect>
            <x>118</x>
            <y>29</y>
            <width>26</width>
            <height>19</height>
           </rect>
          </property>
          <property name="text">
           <string>-</string>
          </property>
         </widget>
         <widget class="QToolButton" name="btnPlus">
          <property name="geometry">
           <rect>
            <x>88</x>
            <y>29</y>
            <width>26</width>
            <height>19</height>
           </rect>
          </property>
          <property name="text">
           <string>+</string>
          </property>
         </widget>
         <widget class="QPushButton" name="btnExpand">
          <property name="geometry">
           <rect>
            <x>12</x>
            <y>29</y>
            <width>72</width>
            <height>20</height>
           </rect>
          </property>
          <property name="text">
           <string>模拟设备</string>
          </property>
         </widget>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QGroupBox" name="QGBSerialConfig">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>171</width>
           <height>211</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>171</width>
           <height>211</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="title">
          <string>串口设置</string>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
         <property name="checkable">
          <bool>false</bool>
         </property>
         <widget class="QPushButton" name="btnOpenPort">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="geometry">
           <rect>
            <x>69</x>
            <y>165</y>
            <width>91</width>
            <height>31</height>
           </rect>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>打开串口</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>:/action/action/centrejust.png</normaloff>
            <normalon>:/action/action/process-stop.png</normalon>:/action/action/centrejust.png</iconset>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
          <property name="autoDefault">
           <bool>false</bool>
          </property>
          <property name="default">
           <bool>false</bool>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QPushButton" name="btnRefresh">
          <property name="geometry">
           <rect>
            <x>11</x>
            <y>165</y>
            <width>51</width>
            <height>31</height>
           </rect>
          </property>
          <property name="toolTip">
           <string>刷新端口号</string>
          </property>
          <property name="text">
           <string>刷新</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>:/action/action/reload.png</normaloff>:/action/action/reload.png</iconset>
          </property>
         </widget>
         <widget class="QWidget" name="">
          <property name="geometry">
           <rect>
            <x>11</x>
            <y>21</y>
            <width>146</width>
            <height>136</height>
           </rect>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="lblPortNum">
             <property name="text">
              <string>端口号</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="cbbPortNum">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maxVisibleItems">
              <number>20</number>
             </property>
             <property name="sizeAdjustPolicy">
              <enum>QComboBox::AdjustToContents</enum>
             </property>
             <property name="minimumContentsLength">
              <number>0</number>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="lblBaudRate">
             <property name="text">
              <string>波特率</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="cbbBaud">
             <property name="sizeAdjustPolicy">
              <enum>QComboBox::AdjustToContents</enum>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="lblDataBit">
             <property name="text">
              <string>数据位</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QComboBox" name="cbbDataBit">
             <property name="sizeAdjustPolicy">
              <enum>QComboBox::AdjustToContents</enum>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="lblVerify">
             <property name="text">
              <string>校验</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QComboBox" name="cbbVerify">
             <property name="sizeAdjustPolicy">
              <enum>QComboBox::AdjustToContents</enum>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="lblStopBit">
             <property name="text">
              <string>停止位</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QComboBox" name="cbbStopBit">
             <property name="sizeAdjustPolicy">
              <enum>QComboBox::AdjustToContents</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
      <zorder>groupBox_8</zorder>
      <zorder>QGBSerialConfig</zorder>
      <zorder>groupBox_2</zorder>
      <zorder>splitter</zorder>
      <zorder>txtOtherValue</zorder>
      <zorder>btnOtherSend</zorder>
      <zorder>splitter_3</zorder>
      <zorder>txtMessage</zorder>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionHelp"/>
   <addaction name="actionModbusPro"/>
  </widget>
  <action name="actionModbusPro">
   <property name="text">
    <string>Modbus协议</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="text">
    <string>帮助说明</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>cbbPortNum</tabstop>
  <tabstop>cbbBaud</tabstop>
  <tabstop>cbbDataBit</tabstop>
  <tabstop>cbbVerify</tabstop>
  <tabstop>cbbStopBit</tabstop>
  <tabstop>btnRefresh</tabstop>
  <tabstop>btnOpenPort</tabstop>
  <tabstop>btnExpand</tabstop>
  <tabstop>btnPlus</tabstop>
  <tabstop>btnSubtract</tabstop>
  <tabstop>checkBox</tabstop>
  <tabstop>robMaster</tabstop>
  <tabstop>robSlaver</tabstop>
  <tabstop>txt10SlaveAddr</tabstop>
  <tabstop>txt10RegAddr</tabstop>
  <tabstop>txt10RegNum</tabstop>
  <tabstop>txt10Value</tabstop>
  <tabstop>btn10Send</tabstop>
 </tabstops>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="bgpMSS"/>
 </buttongroups>
</ui>
